### PR TITLE
Emp 19630

### PR DESCRIPTION
**Pull request by udayasri**

### CODE REVIEW

File: Sources/iOSClientExposure/Entitlements/Parameters/AdsOptions.swift [hunk 1]
```diff
@@ -164,15 +164,13 @@ public struct AdsOptions {
...
-            // returnString["deviceModel"] = UIDevice.current.appleTVModel
+            returnString["deviceModel"] = UIDevice.current.appleTVModel
         }
         
         if let deviceModelNumber = deviceModelNumber {
             returnString["deviceModelNumber"] = deviceModelNumber
         } else {
-            // Should not set by the SDK at the moment
-            // returnString["deviceModelNumber"] = UIDevice.current.appleTVModelNumber
+            returnString["deviceModelNumber"] = UIDevice.current.appleTVModelNumber
```
Comment (shobaki) [inline Sources/iOSClientExposure/Entitlements/Parameters/AdsOptions.swift:173]: No, this should not be set. It won't be necessary for any of SSAI or manifest filtering.

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift [hunk 2]
```diff
@@ -49,54 +49,53 @@ extension UIDevice {
...
-            case "AppleTV3,3": return "A1469"
-            case "AppleTV5,3", "AppleTV6,2": return "A1842"
-            case "AppleTV5,3": return "A1625"
-            case "AppleTV6,2": return "A1844"
-            case "AppleTV11,1": return "A2169"
-            case "AppleTV11,2", "AppleTV11,3": return "A2843"
-            default: return "UNKNOWN"
-            }
-        } */
+    var appleTVModelNumber: String? {
```
[the remark this answers is not in the document]
Comment (udayasri) [reply Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift]: noted. fixed it

File: Sources/iOSClientExposure/Entitlements/Parameters/AdsOptions.swift (hunk 1 above)
Comment (udayasri) [reply Sources/iOSClientExposure/Entitlements/Parameters/AdsOptions.swift]: Noted. fixed it

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift [hunk 3]
```diff
@@ -22,4 +22,80 @@ extension UIDevice {
...
+        case "AppleTV5,3": return "A1625"
+        case "AppleTV6,2": return "A1842"
+        case "AppleTV6,2": return "A1844"
+        case "AppleTV11,1": return "A2169"
+        case "AppleTV11,2", "AppleTV11,3": return "A2843"
+        default: return "UNKNOWN"
+        }
+    } */
+    
+    var appleTVAIdentifier: String? {
```
Comment (bwallberg) [inline Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:86]: un-used?

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift [hunk 4]
```diff
@@ -22,4 +22,80 @@ extension UIDevice {
         if systemName == "iPhone OS" { return "iOS" }
         return systemName
     }
+    
+    var modelName: String {
```
Comment (bwallberg) [inline Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:26]: un-used?

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift [hunk 5]
```diff
@@ -22,4 +22,80 @@ extension UIDevice {
...
+        }
+        switch identifier {
+        case "AppleTV1,1": return "atv-1"
+        case "AppleTV2,1": return "atv-2"
+        case "AppleTV3,1", "AppleTV3,2", "AppleTV3,3": return "atv-3"
+        case "AppleTV5,3": return "atv-4"
+        case "AppleTV6,2": return "atv-4k-1"
+        case "AppleTV11,1": return "atv-4k-2"
+        case "AppleTV11,2", "AppleTV11,3": return "atv-4k-3"
+        default: return "atv-unknown"
```
Comment (bwallberg) [inline Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:59]: Seems to me that you could make an algorithm around this instead.
basically lowercase the string and then split it on the first digit and replace `,` with `-`.
So the deviceModel becomes `appletv-11-1` the mapping to 4k isn't necessary imho .

Comment (udayasri) [reply Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:59]: I just followed the format of the confluence documentation , if we all agree to the pass this format `appletv-11-1` I can fix it.  I agree hardcoding values are not the optimal in this case & would be nice to pass a format that we dynamically create.

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift (hunk 4 above)
Comment (udayasri) [reply Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:26]: Yes it is unused , I will remove this & refactor the extension

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift (hunk 5 above)
Comment (udayasri) [reply Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift]: Updated the PR :)

File: Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift [hunk 6]
```diff
@@ -23,67 +23,9 @@ extension UIDevice {
...
-        case "AppleTV6,2": return "A1844"
-        case "AppleTV11,1": return "A2169"
-        case "AppleTV11,2", "AppleTV11,3": return "A2843"
-        default: return "UNKNOWN"
-        }
-    } */
     
-    var appleTVAIdentifier: String? {
+    /// Return apple device model in formatted way : ex - appletv-11-1 , iphone-12-5 etc
+    var appleDeviceModel: String {
```
Comment (bwallberg) [inline Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift:28]: 😍 I love it

**Review by bwallberg — APPROVED** [review]
Don't forget to update https://tools.cloud.ebms.ericsson.net/confluence/pages/viewpage.action?spaceKey=INTDOC&title=Device+specific+manifest+filtering with the new data :)

File: Sources/iOSClientExposure/Entitlements/Request/PlayEnigmaAsset.swift [hunk 7]
```diff
@@ -61,6 +68,22 @@ public struct PlayEnigmaAsset: ExposureType {
         if let materialProfile = materialProfile {
             parameters["materialProfile"] = materialProfile
         }
+
```
Comment (shobaki) [inline Sources/iOSClientExposure/Entitlements/Request/PlayEnigmaAsset.swift:71]: Just a thought, most code in this struct is common with PlayEnigmaAdsAsset, so a lot of repetition could be avoided, especially the "parameters" part is identical in two places.

Is there a specific reason why struct is used and not class, considering the latter would have been helpful with inheritance / code sharing in this case.

**Review by shobaki — APPROVED** [review]
Looking good to me 👍

**Comment by shobaki** [comment]
> 
> 
> Don't forget to update https://tools.cloud.ebms.ericsson.net/confluence/pages/viewpage.action?spaceKey=INTDOC&title=Device+specific+manifest+filtering with the new data :)

Please also add a note, in that same doc (lower table), about the API or method used to identify make and model. Eg. `systemInfo.machine['identifier']`?

**Comment by iamramanjot** [comment]
LGTM

**Review by iamramanjot — APPROVED** [review]

**Review by bwallberg — APPROVED** [review]
LGTM

**Pull request merged**